Add/Update HTTP downloads algorithms (#9337)
qgis/QGIS-Documentation: docs/user_manual/processing_algs/qgis/filetools.rst

.. _qgisfiledownloader:

Download file via HTTP(S)
----------------------------

> Downloads a URL to the file system with an HTTP(S) GET or POST request.

Reviewer: ```suggestion
Downloads a file specified using a URL with an HTTP(S) GET or POST request.
```

Author: hmm... This was somehow the previous description, and I'm now just following the new description in[ the GUI ](https://github.com/qgis/QGIS/pull/58464/files#diff-ec7018a13f8dcccb8dea5575e46dc6948c744e107ca27856bb31417c2f8ac305R43)

Reviewer: Ah yes. Then leave it. The GUI text was reviewed by a native english speaker (Nyall) so I guess it's right.

------------------------
Performs a HTTP(S) POST/GET request and returns the HTTP status code and the reply data.
If an error occurs then the error code and the message will be returned.
Optionally, the result can be written to a file on disk.
By default the algorithm will warn on errors. Optionally, the algorithm can be set to treat HTTP errors as failures.

Parameters
..........

.. list-table::
   :header-rows: 1
   :widths: 20 20 20 40

   * - Label
     - Name
     - Type
     - Description
   * - **URL or file path**
     - ``URL``
     - [string]
     - The URL or path of the file to open.
   * - **Method**
     - ``METHOD``
     - [enumeration]

       Default: 0
     - The HTTP method to use for the request. Options are:

       * 0 --- GET
       * 1 --- POST
   * - **POST Data**

       Optional
     - ``DATA``
     - [string]

     - The data to add in the body if the request is a POST.
   * - **Authentication**

       Optional
     - ``AUTH_CONFIG``
     - [authconfig]

       Default: No authentication
     - An authentication configuration to pass
   * - **Consider HTTP errors as failures**
     - ``FAIL_ON_ERROR``
     - [boolean]

       Default: False

>      - If set, the algorithm will fail on encountering a HTTP error.

Reviewer: ```suggestion
     - If set, the algorithm will fail on encountering an HTTP error.
```
Not sure...
